### PR TITLE
Add oembed info for mobile

### PR DESCRIPTION
**Pull request by dkasper**
Adding embed info for mobile to the LiveUpdates. Initially I know we thought this was going to cause some issues, however a few things pushed me in this direction.
1) If we don't do this then we will have to fetch embed info for all of the items in the listings... which is a lot.
2) The oembed response is a lot smaller than the extract response and has what we need so should perform well, we were also already using it in the other scraper code.
3) I figured out how to write the scraper code :)

### CODE REVIEW

**Comment by dkasper** [comment]
👓 @MelissaCole @bsimpson63 @KeyserSosa

File: reddit_liveupdate/media_embeds.py [hunk 1]
```diff
@@ -129,6 +145,41 @@ def for_url(cls, url, autoplay=False, maxwidth=_EMBED_WIDTH):
...
+    def matches(cls, url):
+        return True
+
+    def __init__(self, url):
+        super(_LiveEmbedlyScraper, self).__init__(
+            url,
+            maxwidth=500,
+        )
+
+        self.allowed_oembed_types = {"video", "rich", "link", "photo"}
```
Comment (bsimpson63) [inline reddit_liveupdate/media_embeds.py]: can this be a class attribute

File: reddit_liveupdate/media_embeds.py [hunk 2]
```diff
@@ -129,6 +145,41 @@ def for_url(cls, url, autoplay=False, maxwidth=_EMBED_WIDTH):
 
         return _EmbedlyCardFallbackScraper(url, scraper)
 
+# mostly lifted from the EmbedlyScraper in r2
+class _LiveEmbedlyScraper(_OEmbedScraper):
+    OEMBED_ENDPOINT = "https://api.embed.ly/1/oembed"
+
+    @classmethod
+    def matches(cls, url):
+        return True
```
Comment (bsimpson63) [inline reddit_liveupdate/media_embeds.py]: does this mean we'll attempt to scrape every url?

Comment (dkasper) [inline reddit_liveupdate/media_embeds.py]: Yeah, should attempt to scrape any url for the mobile embeds. If there are some cases that we don't want to handle could add some conditions here but I don't know of any right now.

File: reddit_liveupdate/media_embeds.py (hunk 1 above)
Comment (dkasper) [inline reddit_liveupdate/media_embeds.py]: It could but I am overriding a superclass property here

File: reddit_liveupdate/media_embeds.py [hunk 3]
```diff
@@ -253,12 +304,10 @@ def _handle_q(msg):
...
             return
 
-        if not liveupdate.media_objects:
-            return
-
         payload = {
             "liveupdate_id": "LiveUpdate_" + d['liveupdate_id'],
-            "media_embeds": liveupdate.embeds
+            "media_embeds": liveupdate.embeds,
+            "mobile_embeds": liveUpdate.mobile_embeds,
```
Comment (bsimpson63) [inline reddit_liveupdate/media_embeds.py]: upper case U?

**Comment by bsimpson63** [comment]
This isn't isolated to iOS right? So this will update the response for all of liveupdate? Are there any issues with doing that?

**Comment by dkasper** [comment]
That's correct, this is also for Android. We want to support all live events so I think it has to be this way

File: reddit_liveupdate/media_embeds.py (hunk 3 above)
Comment (dkasper) [inline reddit_liveupdate/media_embeds.py]: Whoops, good catch

**Comment by bsimpson63** [comment]
:fish:

**Pull request merged**